typespec-extension/src/code-model-builder.ts

```diff
@@ -896,7 +896,8 @@ export class CodeModelBuilder {
   }
 
   private processParameter(op: CodeModelOperation, param: HttpOperationParameter, clientContext: ClientContext) {
-    if (isApiVersion(this.sdkContext, param)) {
+    if (clientContext.apiVersions && isApiVersion(this.sdkContext, param)) {
+      // pre-condition for "isApiVersion": the client supports ApiVersions
```

Comment:
Major change is here.

If the client is not versioned, do not take the result of "isApiVersion". Make it a normal parameter.